fdbserver/ConsistencyScan.actor.cpp

```diff
@@ -0,0 +1,1090 @@
+/*
+ * ConsistencyScan.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/WorkerInterface.actor.h"
+#include "flow/IRandom.h"
+#include "flow/IndexedSet.h"
+#include "fdbrpc/FailureMonitor.h"
+#include "fdbrpc/Smoother.h"
+#include "fdbrpc/simulator.h"
+#include "fdbclient/DatabaseContext.h"
+#include "fdbclient/ReadYourWrites.h"
+#include "fdbclient/TagThrottle.actor.h"
+#include "fdbserver/Knobs.h"
+#include "fdbserver/StorageMetrics.h"
+#include "fdbserver/DataDistribution.actor.h"
+#include "fdbserver/RatekeeperInterface.h"
+#include "fdbserver/ServerDBInfo.h"
+#include "fdbserver/WaitFailure.h"
+#include "fdbserver/TesterInterface.actor.h"
+#include "flow/DeterministicRandom.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+struct ConsistencyScanData {
```

Comment:
Since a lot of this logic is also in the consistency check worker, could you write a comment block about how we could refactor the workload check to share some code with the scan here?

Reply:
All the common consistency check logic now exists in ConsistencyScan.actor.cpp. It's shared between Consistency Scan and the Consistency Check workload. So there is no duplication. 

I can write some comments to clarify.